### DISCUSSION TITLE
Not able to get Admin console using Keycloak 26.0.6

### DISCUSSION THREAD

**Discussion by pranchalmishra11** [keycloak/keycloak#35477] [Archive]
Hi, i am completely new to this Keycloak, i have installed keycloak latest version in my local windows machine with quarkus distribution, using the following command-: kc.bat start-dev, i started the server, also added the following ENV variables with values against my account user in the windows system-: 

KC_BOOTSTRAP_ADMIN_USERNAME=admin
KC_BOOTSTRAP_ADMIN_PASSWORD=admin123

But still the issue is that when i m hittng the localhost:8080/ end point it is asking me to create temporary administrative user-: following is the message i see on web page-:

You will need local access to create the temporary administrative user.

To create the temporary administrative user open http://localhost:8080/, or set the environment variables KC_BOOTSTRAP_ADMIN_USERNAME and KC_BOOTSTRAP_ADMIN_PASSWORD when starting the server.

Can anyone point out where i am going wrong?

**Comment by shawkins** [comment]
If the server is started for the first time without setting the bootstrap environement variables, the database will be initialized without creating the bootstrap admin. They have to be set for the first time a start command is run.

**Reply by GonzaloGuerraB** [reply]
Thank you very much, this comment has been very helpful to me 👍

**Comment by bzoltowski** [comment]
I encountered this same problem.
The solution for me was to create environment variable "KC_BOOTSTRAP_ADMIN_PASSWORD", with password as value.
Next step was to run command like below: 
kc.bat bootstrap-admin user --username=admin --password:env=KC_BOOTSTRAP_ADMIN_PASSWORD

In logs you should find message like "INFO  [org.keycloak.services] (main) KC-SERVICES0077: Created temporary admin user with username admin"
Then launch keycloak dev (kc.bat start-dev), and you should be able to login to kc over http://localhost:8080/

**Reply by shawkins** [reply]
This is just the expected behavior of the bootstrap-admin command. What's not clear from your description or the original one is why the environment variable seems not to be picked up for the initial start of the server - were the appropriate environment vairables set the first time you started the server from a clean database state?

**Reply by bzoltowski** [reply]
"What's not clear from your description or the original one is why the environment variable seems not to be picked up for the initial start of the server "
Tbh, I have no idea, why environment variables are not picked up for the initial start of the server. - Mby someone from keycloak will reply to this discussion. I just wanted to describe how I solved this problem

"were the appropriate environment vairables set the first time you started the server from a clean database state?" 
Yep, I started the server from a clean database state.
I only downloaded keycloak (26.0.6) and OpenJDK (21).
then at the first try of run kc I got issue with variable JAVA_HOME*,
on 2nd run, I got issue with variables KEYCLOAK_ADMIN* & KEYCLOAK_ADMIN_PASSWORD*,
3rd, I got this one with bootstrap.
when I use mentioned command from provious message, I was able to login to kc.

*when the message with missing variables appear in logs, I setup them as environment variables

**Reply by shawkins** [reply]
> Mby someone from keycloak will reply to this discussion.

That's me :)

> on 2nd run, I got issue with variables KEYCLOAK_ADMIN* & KEYCLOAK_ADMIN_PASSWORD*,

What issue? A deprecation notice? Those are expected to continue to work until support is removed.

**Reply by bzoltowski** [reply]
> That's me :)

Sorry, I didn't know :P

>What issue? A deprecation notice? Those are expected to continue to work until support is removed.

Well, when I started kc and go to website localhost:8080, I got message like this:
'You will need local access to create the temporary administrative user.

To create the temporary administrative user open http://localhost:8080/, or set the environment variables KEYCLOAK_ADMIN and KEYCLOAK_ADMIN_PASSWORD when starting the server.'

funny thing - there were no fields/buttons to create new acc for admin. 
So I setup the environment variables and it solve that issue.

**Reply by HuttTheJAVA** [reply]
thanks! 👍

**Comment by shawkins** [comment]
"To create the temporary administrative user open http://localhost:8080/, or set the environment variables KEYCLOAK_ADMIN and KEYCLOAK_ADMIN_PASSWORD when starting the server."

I don't believe there was any release of keycloak that had this combination of wording. Once the message said temporary, the variables referenced the KC_BOOTSTRAP_ADMIN ones. Are you doing something with a custom theme?

An issue is needed here as the suggestion even in main to use the env variables won't work - it is too late because they are only used when the master realm is created, which would have already happened.

**Comment by chaithuprsnl** [comment]
I’ve been facing the same issue for the past week. In older tutorials, I noticed the use of the KEYCLOAK_ADMIN variable for setting up the admin credentials. However, in the latest Keycloak version (26.0.7), the documentation specifies using KC_BOOTSTRAP_ADMIN_USERNAME=<username> and KC_BOOTSTRAP_ADMIN_PASSWORD=<password> when starting the server.

![ADMIN_Env_Issues](https://github.com/user-attachments/assets/556342ab-f8fe-48fe-9c9a-e538b33665cb)


Here’s the command I am using on a Windows machine:
**cmd /C "set KC_BOOTSTRAP_ADMIN_USERNAME=admin && set KC_BOOTSTRAP_ADMIN_PASSWORD=admin && C:\eMAROffline\keycloak\bin\kc.bat start-dev --log=console --log-console-level=info"**

Also I have seen this in the log - _2024-12-09 07:03:44,025 INFO  [org.keycloak.services] (main) KC-SERVICES0050: Initializing master realm
2024-12-09 07:03:46,412 INFO  [org.keycloak.services] (main) KC-SERVICES0077: Created temporary admin user with username admin_

Still the credentials are not working, it says invalid username or password.

**Reply by chaithuprsnl** [reply]
The issue is resolved. The problem is with the command I used - cmd /C "set KC_BOOTSTRAP_ADMIN_USERNAME=admin && set KC_BOOTSTRAP_ADMIN_PASSWORD=admin && C:\eMAROffline\keycloak\bin\kc.bat start-dev --log=console --log-console-level=info". As I am setting environment variables and running keycloak in same command. The variables are not yet visible to the keycloak. I used this command instead & it is working fine. I am able to login to keycloak console.

kc.bat start-dev --bootstrap-admin-username admin --bootstrap-admin-password admin123

**Comment by itsmesangram5** [comment]
Try using 127.0.0.1:8080 instead of localhost:8080

**Reply by abdoochabrik** [reply]
the correct one

**Reply by lifes-a-bitch** [reply]
makes a big difference :-) it worked.
Thanks a lot

**Reply by khaldi-yass** [reply]
What a weird and unconventional solution for something that is supposedly the first step of every keycloak admin, thanks for your input.

**Comment by kalpanaManjunath** [comment]
I faced same issue  with keycloak 26.0.7 while setting it on RHEL server. This issue chain here helped me.  As suggested, I removed existing keycloak folder, unziped another set and executed below commands. 
export KC_BOOTSTRAP_ADMIN_USERNAME=admin
export KC_BOOTSTRAP_ADMIN_PASSWORD=pwd

kc.sh start-dev --bootstrap-admin-username $KC_BOOTSTRAP_ADMIN_USERNAME --bootstrap-admin-password $KC_BOOTSTRAP_ADMIN_PASSWORD
This worked, then logged with admin and pwd .

**Reply by JackTor** [reply]
That worked for me!!! thank you

**Comment by makampos** [comment]
I ran the same issue, however, in my case, I was attempting to access it using the docker-compose file.

These are the environment variables in my docker-compose.yml :

```
environment:
 - KEYCLOAK_ADMIN=admin
 - KEYCLOAK_ADMIN_PASSWORD=admin
```

Then I exported them using the following command in my terminal :

```
export KEYCLOAK_ADMIN=admin
export KEYCLOAK_ADMIN_PASSWORD=admin
```

**Just so you know, you need to set the environment variable before spinning up the keycloak container**, because the credentials will be set during the container first initialization.

Once the container is fully operational, the logs should show you the following result.

`[org.keycloak.services] (main) KC-SERVICES0009: Added user 'admin' to realm 'master'`

This is the version I'm currently using `quay.io/keycloak/keycloak:24.0.3` and running docker desktop on a macOS.

**Comment by lamda-cdm** [comment]
Heyy , so i faced a similar thing . What i did was delete the keycloak folder and then set the environment variables for admin and password.
I then downloaded keycloak again and started the server. It created the temporary admin .
Look for the log 
**KC-SERVICES0077: Created temporary admin user with username admin**

**Reply by Iykechuks11** [reply]
Greetings superheroes!!! 

Who resolved this on ubuntu?

**Reply by Iykechuks11** [reply]
![image](https://github.com/user-attachments/assets/493d0e71-9d9d-41e0-898c-3e897f5d1eab)

**Reply by Iykechuks11** [reply]
I get this error: 


WARN  [org.keycloak.events] (executor-thread-2) type="LOGIN_ERROR", realmId="fbc084de-1176-4c33-9176-78af4b6db462", realmName="master", clientId="null", userId="null", ipAddress="185.100.244.59", error="ssl_required"

**Reply by snettar** [reply]
Hi @Iykechuks11 ,

It looks like your realm "master" is expecting all connections to be done via https only.  
For more details on this aspect, you can refer this https://www.keycloak.org/docs/latest/server_admin/#sslhttps-requirement which then takes you to https://www.keycloak.org/docs/latest/server_admin/#_ssl_modes

Assuming you are doing this in development, if you want to access UI , then you can configure your master realm to use "none" as your Require SSL mode. This should probably allow you to access the Keycloak GUI.

**Comment by manojsathyanarayana** [comment]
I pulled bitnami/keycloak:26.2.5-debian-12-r3 and customized it by injecting the internal libraries. The build was success in Docker and also in Docker compose.
Here are the details of the images and charts.
Keycloak image: bitnami/keycloak:26.2.5-debian-12-r3
Keycloak's postgres image: 15.4.0-debian-11-r45
Bitnami keycloak chart version: 24.7.4
EKSversion: 1.29

Error: 

![image](https://github.com/user-attachments/assets/384ce05b-4a41-486f-bdf6-4620a4ce9b4a)

And facing issues with the admin page login. Please suggest.

**Reply by shawkins** [reply]
@pranchalmishra11 please start a new discussion. More than likely this is a configuration issue.

**Comment by stefantul** [comment]
In my case, I managed to solve the login issue by escaping the $ character in the password variable in the .env file. I'm deploying Keycloak in a docker container via docker compose. When inspecting the container environment variables I noticed that
`KEYCLOAK_PASSWORD=abcdef$fghij` was being registered in the docker container as `KEYCLOAK_PASSWORD=abcdef$`. 

The solution is to escape the $ character such as `\$`

**Comment by NinoBach** [comment]
I was unable to log into the admin console of keycloak. I've checked the environment variables and was confused, because they existed, just like they did in another working deployment. Got this error: invalid_user_credentials

It turns out that I had the character "ö" in a password, which is a non-ascii character. I removed that character and was able to login afterwards.

**Comment by mealibek** [comment]
Hello.
Just run this command below

`docker run -p 8080:8080 quay.io/keycloak/keycloak start-dev --bootstrap-admin-username admin --bootstrap-admin-password admin_password`
